### ISSUE TITLE
Code lens for .sql files and/or inlined sql fences

### ISSUE THREAD

**Issue by archiewood** [evidence-dev/evidence-vscode#45]
Aware that this is a "next week" (rather than right now) task but wanted somewhere to write about this.

Feel free to break down into smaller issues and close this one if it makes sense.

## Draft Acceptance Criteria (to be refined)
- User can run [SQL file queries](https://docs.evidence.dev/core-concepts/queries/#sql-file-queries) in VSCode and get results into some kind of visual output
- Works for any [supported database adapter](https://docs.evidence.dev/core-concepts/data-sources/#supported-data-sources)
- Works for [chained queries](https://docs.evidence.dev/core-concepts/queries/#query-chaining)
- (Optional) User can run [inline queries](https://docs.evidence.dev/core-concepts/queries/#writing-queries) in VSCode. Results are returned in table ?below the markdown fence in the code
- (Optional) User gets autocomplete for their database. @RandomFractals I think you suggested we might be able to rely on another extension for this. If this is hard / time consuming then deprioritize.
- (Very optional) User can see tables, schemas etc in their database in some kind of sidebar. I have no idea how hard this is, and may want to deal with separately, possibly in the future.

## Other thoughts
- We should try to sensibly handle if a user tries to `select * from table` and return 100,000 rows to VSCode. Should we break, return error, paginate

**Comment by RandomFractals** [comment]
We'll take your notes and design ideas for this feature into consideration.

However, this is not the limited initial set of SQL Code Lens feature that is in scope and to be delivered next week.

Let's draft the Acceptance Criteria after we actually define the target feature details, SQL Code Lens and tabular data results display in this case.

**Comment by archiewood** [comment]
Gotcha, have we got that scope written down anywhere yet?

**Comment by RandomFractals** [comment]
yes, we have the scope defined in our 3 week project SOW, notes, confirmation and review of that scope from this morning in the related slack channel for this project I can't link to because it's private.

This feature will be defined and reviewed during our Sprint 3 planning meeting we'll kick off next week.

What you have above is a very generic wish list, with some desirable options/UI features that lack the detail, or related tools info and integration options and descriptions that will be outlined for Sprint 3 in a separate feature ticket.

For those reasons, I will close this ticket for now, so that we don't set inappropriate and vague expectations of what will be included in the initial set of this feature.

Thank you for providing links to related documents in the Evidence app. I will use them and review the related code as well.
